### PR TITLE
Improve the migrate process

### PR DESCRIPTION
**Pull request by stonezdj**
In order to separate the migrate process from core process, add parameter -mode for main function, it can be normal, migrate, skip_migrate.
Signed-off-by: stonezdj <stonezdj@gmail.com>

### CODE REVIEW

**Review by chlins — APPROVED** [review]
lgtm

File: src/core/main.go [hunk 1]
```diff
@@ -202,9 +206,23 @@ func main() {
...
+		if err = migration.Migrate(database); err != nil {
+			log.Fatalf("failed to migrate the database, error: %v", err)
+			os.Exit(1)
+		}
+		log.Info("the database migrate success")
+		os.Exit(0)
+	} else if strings.EqualFold(*runMode, "skip_migrate") {
+		log.Info("skip the database migrate")
+	} else {
+		// Run migrator without exit, work as normal
```
Comment (wy65701436) [inline src/core/main.go:220]: it will be exit with rc 1 on normal failure, please revise the comments.

File: src/core/main.go [hunk 2]
```diff
@@ -202,9 +206,23 @@ func main() {
 	if err := dao.InitDatabase(database); err != nil {
 		log.Fatalf("failed to initialize database: %v", err)
 	}
-	if err = migration.Migrate(database); err != nil {
-		log.Fatalf("failed to migrate: %v", err)
+	if strings.EqualFold(*runMode, "migrate") {
+		// Used by Harbor helm preinstall, preupgrade hook container
+		if err = migration.Migrate(database); err != nil {
+			log.Fatalf("failed to migrate the database, error: %v", err)
+			os.Exit(1)
```
Comment (wy65701436) [inline src/core/main.go:213]: log.Fatalf calls the os.Exit(1) eventually, please remove it.

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/core/main.go (hunk 2 above)
Comment (stonezdj) [reply src/core/main.go:213]: updated

File: src/core/main.go [hunk 3]
```diff
@@ -114,6 +115,9 @@ func gracefulShutdown(closing, done chan struct{}, shutdowns ...func()) {
 }
 
 func main() {
+	runMode := flag.String("mode", "normal", "The harbor-core container run mode, it could be normal, migrate or skip_migrate, default is normal")
```
Comment (reasonerjt) [inline src/core/main.go:118]: This is trivial but it's more common the parm uses `-` as separator if the subcommand has multiple words?
For example 
`kubectl port-forwarding` , `kubectl cluster-info`

Comment (stonezdj) [reply src/core/main.go]: Updated

**Pull request merged**